thunder/executors/pythonex.py

```diff
@@ -226,7 +226,7 @@ def _clear_collection_meta(coll: CollectionProxy) -> None:
 
 
 def _clear_collection_prim_impl(a: Sequence) -> None:
-    if isinstance(a, list):
+    if isinstance(a, (list, dict)):
```

Comment:
There are [MutableSequence](https://docs.python.org/3/library/collections.abc.html#collections.abc.MutableSequence), [MutableMapping](https://docs.python.org/3/library/collections.abc.html#collections.abc.MutableMapping), and [MutableSet](https://docs.python.org/3/library/collections.abc.html#collections.abc.MutableSet) types that we can probably use here for the isinstance check.